src/OrbitCore/SymbolHelper.cpp

```diff
@@ -128,22 +133,37 @@ std::vector<fs::path> FindStructuredDebugDirectories() {
 
 ErrorMessageOr<void> SymbolHelper::VerifySymbolsFile(const fs::path& symbols_path,
                                                      const std::string& build_id) {
-  OUTCOME_TRY(symbols_file, CreateElfFile(symbols_path));
-
-  if (!symbols_file->HasDebugSymbols()) {
-    return ErrorMessage(
-        absl::StrFormat("Elf file \"%s\" does not contain symbols.", symbols_path.string()));
+  auto object_file_or_error = CreateObjectFile(symbols_path);
+  if (object_file_or_error.has_error()) {
+    return ErrorMessage(absl::StrFormat("Unable to load object file \"%s\": %s",
+                                        symbols_path.string(),
+                                        object_file_or_error.error().message()));
   }
-  if (symbols_file->GetBuildId().empty()) {
-    return ErrorMessage(
-        absl::StrFormat("Symbols file \"%s\" does not have a build id", symbols_path.string()));
+
+  if (object_file_or_error.value()->IsElf()) {
+    ElfFile* symbols_file = dynamic_cast<ElfFile*>(object_file_or_error.value().get());
+    if (!symbols_file->HasDebugSymbols()) {
+      return ErrorMessage(
+          absl::StrFormat("Elf file \"%s\" does not contain symbols.", symbols_path.string()));
+    }
+    if (symbols_file->GetBuildId().empty()) {
+      return ErrorMessage(
+          absl::StrFormat("Symbols file \"%s\" does not have a build id", symbols_path.string()));
+    }
+    const std::string& file_build_id = symbols_file->GetBuildId();
+    if (build_id != file_build_id) {
+      return ErrorMessage(
+          absl::StrFormat(R"(Symbols file "%s" has a different build id: "%s" != "%s")",
+                          symbols_path.string(), build_id, file_build_id));
+    }
+    return outcome::success();
   }
-  const std::string& file_build_id = symbols_file->GetBuildId();
-  if (build_id != file_build_id) {
+
+  if (!object_file_or_error.value()->HasDebugSymbols()) {
     return ErrorMessage(
-        absl::StrFormat(R"(Symbols file "%s" has a different build id: "%s" != "%s")",
-                        symbols_path.string(), build_id, file_build_id));
+        absl::StrFormat("Object file \"%s\" does not contain symbols.", symbols_path.string()));
   }
```

Comment:
nit: seems like this check could go before if and you can remove the one on line 145